flare_portal/api/tests/test_views.py

```diff
@@ -102,6 +103,57 @@ def test_validation(self) -> None:
         self.assertEqual(resp.json(), {"participant": ["Invalid participant"]})
 
 
+class SubmissionAPIViewTest(TestCase):
+    def test_post(self) -> None:
+        experiment: Experiment = ExperimentFactory()
+        participant = ParticipantFactory(
+            participant_id="Flare.ABCDEF", experiment=experiment
+        )
+
+        # Try to finish the experiment
+        submit_resp = self.client.post(
+            reverse("api:submission"),
+            {"participant": "Flare.ABCDEF"},
+            content_type="application/json",
+        )
+
+        # Test that the request fails because it hasn't started
+        self.assertEqual(400, submit_resp.status_code)
+
+        # Start experiment
+        config_resp = self.client.post(
+            reverse("api:configuration"),
+            {"participant": "Flare.ABCDEF"},
+            content_type="application/json",
+        )
+
+        # Test that the participant object has been updated
+        config_data = config_resp.json()
+        updated_participant = Participant.objects.get(pk=participant.id)
+        self.assertEqual(config_data["participant_started_at"], None)
+        self.assertEqual(config_data["participant_finished_at"], None)
```

Comment:
Shouldn't the `participant_started_at` field be filled in at this point?

Reply:
Ah I think I see why you want this to be None. Because you're using this to determine whether or not the participant has logged in before.

I was thinking more along the lines of the API returning a 400 bad request if you try to login a second time with the same participant. That way it's clear that the behaviour is not allowed.

That approach also removes the need to return the start/finish times on the API.